src/Block/GalleryBlockService.php

```diff
@@ -140,7 +145,7 @@ public function buildEditForm(FormMapper $formMapper, BlockInterface $block)
             'translation_domain' => 'SonataMediaBundle',
         ]);
         $fieldDescription->setAssociationAdmin($this->getGalleryAdmin());
-        $fieldDescription->setAdmin($formMapper->getAdmin());
+        $fieldDescription->setAdmin($this->container->get('sonata.page.admin.block'));
```

Comment:
It is OK for old Symfony practice.

For new pratice we should avoid using container. Please move it this service to block property and set it in constructor. This will allow inject required servaices insted container.

Reply:
I think the container should be removed in next major not in this patch. The container is used else where in the blocks too for example when we are getting the media admin. Can we have another pull request for that?

Reply:
Container should be remove in Block 4 support becouse you must change constructor parameters.

Reply:
Well this can be changed to be "prepare for Block 4 support". 
I would happily do this later. 
This was supposed to be patch for these blocks to work at all and now they work.
OK?